runtime/compiler/ilgen/Walker.cpp

```diff
@@ -7012,6 +7186,89 @@ TR_J9ByteCodeIlGenerator::storeFlattenableInstanceWithHelper(int32_t cpIndex)
+    * n6n      (  0)  NULLCHK on n4n [#32]
+    * n5n      (  2)    iloadi  Point2D.x I[#355  final Point2D.x I +4]
+    * n4n      (  2)      aload  <parm 1 Q>[#353  Parm]
+    * n7n      (  0)  istorei  AssortedRefWithLongAlignment.point.x I[#354  final AssortedRefWithLongAlignment.point.x I +28]
+    * n3n      (  2)    aload  <'this' parm LAssortedRefWithLongAlignment;>[#352  Parm]
+    * n5n      (  2)    ==>iloadi
+    * n9n      (  0)  NULLCHK on n4n [#32]
+    * n8n      (  2)    iloadi  Point2D.y I[#357  final Point2D.y I +8]
+    * n4n      (  2)      ==>aload
+    * n10n     (  0)  istorei  AssortedRefWithLongAlignment.point.y I[#356  final AssortedRefWithLongAlignment.point.y I +32]
+    * n3n      (  2)    ==>aload (X!=0 sharedMemory )
+    * n8n      (  2)    ==>iloadi
+    * ---- stack after: empty -----------------
+    */
+   TR_ResolvedJ9Method * owningMethod = static_cast<TR_ResolvedJ9Method*>(_methodSymbol->getResolvedMethod());
+
+   int32_t prefixLen = 0;
+   char * fieldNamePrefix = getTopLevelPrefixForFlattenedFields(owningMethod, cpIndex, prefixLen, comp()->trMemory()->currentStackRegion());
+
+   TR_OpaqueClassBlock * containingClass = owningMethod->definingClassFromCPFieldRef(comp(), cpIndex, _methodSymbol->isStatic());
+   const TR::TypeLayout *containingClassLayout = comp()->typeLayout(containingClass);
+   size_t fieldCount = containingClassLayout->count();
+
+   TR::Node * value = pop();
+   TR::Node * address = pop();
+
+   int len;
+   const char *fieldClassChars = owningMethod->fieldSignatureChars(cpIndex, len);
+   TR_OpaqueClassBlock * fieldClass = fej9()->getClassFromSignature(fieldClassChars, len, owningMethod);
+
+   for (size_t idx = 0; idx < fieldCount; idx++)
+      {
+      const TR::TypeLayoutEntry &fieldEntry = containingClassLayout->entry(idx);
+      if (!strncmp(fieldNamePrefix, fieldEntry._fieldname, prefixLen))
+         {
+         auto * fieldSymRef = comp()->getSymRefTab()->findOrFabricateShadowSymbol(containingClass,
+                                                                     fieldEntry._datatype,
+                                                                     fieldEntry._offset,
+                                                                     fieldEntry._isVolatile,
+                                                                     fieldEntry._isPrivate,
+                                                                     fieldEntry._isFinal,
+                                                                     fieldEntry._fieldname,
+                                                                     fieldEntry._typeSignature
+                                                                     );
+
+         char * fieldNameRemovedTopLevelPrefix = removeTopLevelPrefixForFlattenedFields(fieldEntry._fieldname, prefixLen,
+                                                                                        comp()->trMemory()->currentStackRegion());
```

Comment:
As above, does this actually need to create a copy of the name with the prefix removed or could it simply be written as follows?

```
char *fieldNameRemovedTopLevelPrefix = fieldEntry._fieldname + prefixLen;
```